### PR TITLE
Skip compile external_custom project if EXTERNAL_CUSTOM_REPO is not set

### LINKED ISSUE #3187: Assess the health of external test group
For each of the tests listed in https://github.com/adoptium/aqa-tests/tree/master/external we should assess their current health and status for all versions and implementations.

- [Hotspot spreadsheet](https://docs.google.com/spreadsheets/d/1kcHs1f7H1bmlLvkRBZ0CMdTaGBa3DWEXP86bYRefxg8/edit?usp=sharing)
- [OpenJ9 spreadsheet](https://docs.google.com/spreadsheets/d/197JBPA9hyYDLwc7xNncUKoXiKj9B7BH2Itx8ktdYgE4/edit?usp=sharing)

### PR DESCRIPTION
**Pull request by sophia-guo**
Skip compile external_custom project if EXTERNAL_CUSTOM_REPO is not set
Move external_custom tests to special level

Fix #3187 

Signed-off-by: Sophia Guo <sophia.gwf@gmail.com>

### CODE REVIEW

**Comment by sophia-guo** [comment]
Tested sanity.external: https://ci.adoptopenjdk.net/view/work-in-progress/job/grinder_sandbox_new/436/

File: external/external_custom/playlist.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@
 			<level>extended</level>
 		</levels>
 		<groups>
-			<group>external</group>
+			<group>functional</group>
```
Comment (llxia) [inline external/external_custom/playlist.xml:27]: why do we move the external test to the functional group?

Comment (sophia-guo) [reply external/external_custom/playlist.xml]: Typo.  Updated.

**Review by llxia — APPROVED** [review]
LGTM

**Review by smlambert — APPROVED** [review]
thanks @sophia-guo !

**Pull request merged**